lib/tools/adb-commands.js

```diff
@@ -1608,4 +1609,26 @@ methods.screenrecord = function (destination, options = {}) {
   return new SubProcess(this.executable.path, fullCmd);
 };
 
+/**
+ * Performs the given editor action on the focused input field.
+ * This method requires Appium Settings helper to be installed on the device.
+ * No exception is thrown if there was a failure while performing the action.
+ * You must investigate the logcat output if something did not work as expected.
+ *
+ * @param {string|number} action - Either action code or name. The following action
+ *                                 names are supported: `normal, unspecified, none,
+ *                                 go, search, send, next, done, previous`
+ */
+methods.performEditorAction = async function (action) {
+  log.debug(`Performing editor action: ${action}`);
+  const defaultIME = await this.defaultIME();
+  await this.enableIME(APPIUM_IME);
+  try {
+    await this.setIME(APPIUM_IME);
+    await this.shell(['input', 'text', `/${action}/`]);
+  } finally {
+    await this.setIME(defaultIME);
```

Comment:
do we want to set it back to the default? what if  the user has already set their own ime? maybe we should get the current IME instead? (could also then save some time if for whatever reason it happens to already be the appium IME)

Reply:
actually the default ime is the currently selected one. It is just the confusing name of the function